lib/elixir/test/elixir/gen_server_test.exs

```diff
@@ -50,8 +50,20 @@ defmodule GenServerTest do
     assert GenServer.cast(:foo, {:push, :world}) == :ok
   end
 
+  @tag capture_log: true
   test "call/3 exit messages" do
+    {:ok, pid} = GenServer.start_link(Stack, [:hello])
+
+    {:ok, stoped_pid} = GenServer.start_link(Stack, [:hello])
+    GenServer.stop(stoped_pid)
+
+    {:ok, killed_pid} = GenServer.start(Stack, [:hello], name: :stack)
+    Process.exit(killed_pid, :kill) # Simulating condition when GenServer killed
+    
     assert catch_exit(GenServer.call(self, :pop, 5000)) == {:calling_self, {GenServer, :call, [self, :pop, 5000]}}
+    assert catch_exit(GenServer.call(pid, :pop, 0)) == {:timeout, {GenServer, :call, [pid, :pop, 0]}}
```

Comment:
We should change the GenServer implementation to have a `:noreply` if we are testing time outs and send the call as `:noreply`. If the GenServer replies it is possible for the reply to be in the message queue when the receive occurs.